go/pserver/optimizer.go

```diff
@@ -78,7 +80,11 @@ func (o *optimizer) UpdateParameter(g Gradient) error {
 		return fmt.Errorf("Name: %s, parameter and gradient element type not match, parameter: %v, gradient: %v", g.Name, o.elementType, g.ElementType)
 	}
 
-	r := C.paddle_update_parameter(o.opt, C.paddle_element_type(g.ElementType), unsafe.Pointer(&g.Content[0]), C.int(len(g.Content))/C.sizeof_float)
+	if o.contentLen != len(g.Content) {
+		return fmt.Errorf("Name: %s, parameter and gradient does not have same content len, parameter: %d, gradient: %d", g.Name, o.contentLen, len(g.Content))
+	}
+
+	r := C.paddle_update_parameter(o.opt, C.paddle_element_type(g.ElementType), unsafe.Pointer(&g.Content[0]), C.int(len(g.Content)))
```

Comment:
In `optimizer/optimizer.cc`, Content is casted to float array, and length(num_bytes) is passed to `Tensor.h`
```c++
int paddle_update_parameter(paddle_optimizer* o,
                            const paddle_element_type data_type,
                            const void* grad_buffer,
                            int num_bytes) {
  auto grad_type = reinterpret_cast<const float*>(grad_buffer);
  Tensor* gradient = new Tensor(const_cast<float*>(grad_type), num_bytes);
```
in `optimizer/Tensor.h`:

```c++
TensorT(T* data, size_t size)
      : height_(1), width_(size), data_ptr_(nullptr), data_(data) {}
```

The num_bytes is used as width, which means count of float numbers. So this is not bytes.

Maybe we should change the parameter name of `paddle_update_parameter` `num_bytes` to `array_size`